SignalServiceKit/src/Messages/OWSMessageUtils.h

```diff
@@ -17,6 +21,12 @@ NS_ASSUME_NONNULL_BEGIN
 
 - (void)updateApplicationBadgeCount;
 
++ (nullable TSQuotedMessage *)quotedMessageForIncomingMessage:(TSIncomingMessage *)message
+                                                  transaction:(YapDatabaseReadWriteTransaction *)transaction;
+
++ (nullable TSQuotedMessage *)quotedMessageForOutgoingMessage:(TSOutgoingMessage *)message
+                                                  transaction:(YapDatabaseReadWriteTransaction *)transaction;
```

Comment:
Deriving a message that quotes another message is non-trivial - and we'll do it multiple places, so these are factory methods that DRY up that logic.